src/common/FileSystem.cpp

```diff
@@ -909,6 +909,20 @@ class ZipArchive {
 		return fileInfo.uncompressed_size;
 	}
 
+	// Check if currently open file is a symbolic link
+	bool CheckSymlink(std::error_code& err) const
+	{
+		unz_file_info64 fileInfo;
+		int result = unzGetCurrentFileInfo64(zipFile, &fileInfo, nullptr, 0, nullptr, 0, nullptr, 0);
+		if (result != UNZ_OK) {
+			SetErrorCodeZlib(err, result);
+			return false;
+		}
+		ClearErrorCode(err);
+		uLong attr = fileInfo.external_fa >> 16;
+		return ( ( attr & 0120000 ) == 0120000 );
```

Comment:
Can we have named constants for these magic numbers?

Reply:
yep for the idea, I'm still looking for a proper name for these magic numbers (especially 16 which I not found in Darkplaces…)